src/cpp/modules/manualEigen/lstm_eigen_d_structures.h

```diff
@@ -0,0 +1,274 @@
+#pragma once
+
+#include <vector>
+
+#include "../../shared/lstm_eigen_types.h"
+
+template<typename T>
+struct StateElementGradientModel
+{
+    // Matrix(hsize, 10) derivatives from all layers
+    MapX10<T> d_rawX10;
```

Comment:
Here and below `MapXN` varaiables need explanations in comments: what exactly are they (which derivatives they incorporate) and why do we need them.

Reply:
Comments added.